crates/ide-db/src/path_transform.rs

```diff
@@ -119,19 +126,25 @@ impl<'a> PathTransform<'a> {
             // can still hit those trailing values and check if they actually have
             // a default type. If they do, go for that type from `hir` to `ast` so
             // the resulting change can be applied correctly.
-            .zip(self.substs.types.iter().map(Some).chain(std::iter::repeat(None)))
-            .filter_map(|(k, v)| match k.split(db) {
-                Either::Left(_) => None, // FIXME: map const types too
-                Either::Right(t) => match v {
-                    Some(v) => Some((k, v.ty()?.clone())),
-                    None => {
-                        let default = t.default(db)?;
-                        let v = ast::make::ty(
+            .zip(self.substs.types_and_consts.iter().map(Some).chain(std::iter::repeat(None)))
+            .filter_map(|(k, v)| match (k.split(db), v) {
+                (_, Some(v)) => {
+                    let subst =
+                        TypeOrConstSubst { subst: v.ty()?.clone(), to_be_transformed: false };
+                    Some((k, subst))
+                }
+                (Either::Right(t), None) => {
+                    let default = t.default(db)?;
+                    let subst = TypeOrConstSubst {
+                        subst: ast::make::ty(
                             &default.display_source_code(db, source_module.into(), false).ok()?,
-                        );
-                        Some((k, v))
-                    }
-                },
+                        )
+                        .clone_for_update(),
+                        to_be_transformed: true,
+                    };
+                    Some((k, subst))
+                }
+                (Either::Left(_), None) => None, // FIXME: get default const value
```

Comment:
it seems impossible to get default const values from hir right now; I hope to fix this line after patching hir

see https://rust-lang.zulipchat.com/#narrow/stream/185405-t-compiler.2Frust-analyzer/topic/hir.20.26.20generic.20const.20params/near/365836700